### DISCUSSION TITLE
making the game easier to distribute

### DISCUSSION THREAD

**Discussion by goolord** [ikemen-engine/Ikemen-GO#1243] [Feature Request]
### Is your feature request related to a problem? Please describe.

when I want to play against friends in netplay, my solution to make sure we have the same stages / characters / select screen / sounds etc is to just send them a torrent of my entire copy of ikemen-go. this also usually has the unwanted side effect of giving them my save data, which is necessary for the select screen.

### Describe the solution you'd like

it would be nice if there was some kind of tool for syncing resources between two people, or if there was some kind of principled, non-mugen esque way to define resources like the select screen in an easily distributable fashion. here are ideas that i have

- anything not strictly user configured, like the select screen, can go in a separate file, which if not present will fallback to how mugen parses saves
- ideally when connecting to someone over network, we prompt the user to sync assets like characters and stages and use the network connect to send those, maybe gzipped or something
- it would also be nice if certain assets could be compressed. sounds for example tend to take up a lot of disk space and the majority of them are not read, so it would be nice to be able to gzip them and have them uncompressed on the fly or something.
- also on my wishlist, which i don't know if it needs to be something present in this repo, is a tool that would allow me to patch stuff like `select.def` on my friends' machines so they can have a more up to date character select screen without me having to send them the file and then explain where to put it.

i can somewhat write go, so i'd be happy to contribute if the maintainers are happy with any of these suggestions.

**Comment by potsmugen** [comment]
That sounds like an undertaking almost as big as the engine itself. Could be a separate program maybe.
